File: src/main/java/dev/slne/discord/annotation/ChannelCreationModal.java
package dev.slne.discord.annotation;

import dev.slne.discord.ticket.TicketType;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface ChannelCreationModal {

  String modalId() default "";

  TicketType ticketType();

}

File: src/discordold/annotation/ChannelCreationModal.kt
package dev.slne.discordold.annotation

import dev.slne.discordold.ticket.TicketType

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class ChannelCreationModal(val modalId: String = "", val ticketType: TicketType)
